### ISSUE TITLE
Locating a package in a subdirectory

### ISSUE THREAD

**Issue by blinky3713** [purescript/spago#758]
Is there any way to refer to a subdirectory in a `packages.dhall` entry? My situation is that I want monorepo two packages `A` and `B` that often move together but `B` depends on `A`, and depend on both from another repository `C`.

`A` and `B` are both referenced as `Local` in the monorepo as per the instructions in the README, and that part works fine. However as they are nested in the monorepo, they are nested in `.spago` directory in project `C` and it seems like they can't be found. Ironically at some point before I blew everything in `C` away and rebuilt, i was getting a duplicate module error as I mentioned [here](https://discourse.purescript.org/t/how-do-i-depend-on-multiple-packages-from-a-monorepo-with-spago/2219)

Basically I am looking for something like the `subdirs` option for a `stack.yaml` declaration. [here](https://docs.haskellstack.org/en/stable/yaml_configuration/#packages)

Anyway, let me know if there is some way to do this, or if this shouldn't be allowed because of the design philosophy. Thanks!

**Comment by f-f** [comment]
@blinky3713 could you post the layout of your repo and the content of your `packages.dhall`? I have trouble understanding how things are structured.

**Comment by blinky3713** [comment]
Sure, sorry about that.

This is the monorepo containing two packages, purescript-web3 and purescript-web3-generator:
https://github.com/f-o-a-m/purescript-web3/tree/monorepo

Here is the relevant `packages.dhall` for that monorepo:
https://github.com/f-o-a-m/purescript-web3/blob/monorepo/packages.dhall#L178-L180

Here is the `packages.dhall` file for the project that depends on the monorepo's libraries:
https://github.com/f-o-a-m/chanterelle/blob/monorepo-web3/packages.dhall#L7-L58

Here is what is pulled into `.spago/web3/monorepo` `.spago/web3-generator/monorepo`  and  after running `spago build`:
<img width="847" alt="Screen Shot 2021-03-24 at 11 49 11 AM" src="https://user-images.githubusercontent.com/31280145/112367274-f5b20600-8c96-11eb-9c11-fc53f00b4645.png">

Which is what I meant by the nesting.

**Comment by blinky3713** [comment]
@f-f Just checking in, any thoughts about this?

**Comment by f-f** [comment]
@blinky3713 sorry, I'm travelling until next week and I didn't manage to have a look at this yet, will do ASAP

**Comment by wclr** [comment]
@blinky3713 

Though I think it is supported what you need, but it really not clear what goes wrong. You should try to give a more structured explanation for the case and the problem.

**Comment by blinky3713** [comment]
@wclr I'm not sure I can really describe the problem much better than what's in the comment above starting with 

> Sure, sorry about that.

And the error message is that the modules from the `web3` and `web3-generator` packages can't be found when trying to build `chanterelle`.

The most succinct way I can put it for haskell/stack users is what I said above:
> Basically I am looking for something like the subdirs option for a stack.yaml declaration. here

**Comment by wclr** [comment]
In chanterelle (in its package.dhall) project packages `web3` and `web3-generator` are installed from **the same** git repo  https://github.com/f-o-a-m/purescript-web3

What is there about subfolders?

**Comment by f-f** [comment]
I finally had some time to look at this - indeed subdirectories are not supported right now.
The main reason is that there's an ecosystem-wide assumption about library sources being in `src/**/*.purs`. This was discussed in #288 (and this question pops up often, e.g. #755).
We're working to lift this limitation in the new Registry, more discussion about that in purescript/registry#16

**Comment by blinky3713** [comment]
Thanks for getting back to me! I will try to catch myself up to speed on those issues to see what remains. As a general note it's been extremely pleasant to update all of our repos to spago, very much appreciating all of your hard work!

**Comment by f-f** [comment]
@blinky3713 thank you! :heart: 

I think there are no more actionable points in this issue, so I'll close this :slightly_smiling_face:
